protocol/communities/community_event_message.go

```diff
@@ -0,0 +1,257 @@
+		if communityEvent.TokenPermission == nil || len(communityEvent.TokenPermission.Id) == 0 {
+			return errors.New("invalid token permission delete event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_CATEGORY_CREATE:
+		if communityEvent.CategoryData == nil || len(communityEvent.CategoryData.CategoryId) == 0 {
+			return errors.New("invalid community category create event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_CATEGORY_DELETE:
+		if communityEvent.CategoryData == nil || len(communityEvent.CategoryData.CategoryId) == 0 {
+			return errors.New("invalid community category delete event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_CATEGORY_EDIT:
+		if communityEvent.CategoryData == nil || len(communityEvent.CategoryData.CategoryId) == 0 {
+			return errors.New("invalid community category edit event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_CHANNEL_CREATE:
+		if communityEvent.ChannelData == nil || len(communityEvent.ChannelData.ChannelId) == 0 ||
+			communityEvent.ChannelData.Channel == nil {
+			return errors.New("invalid community channel create event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_CHANNEL_DELETE:
+		if communityEvent.ChannelData == nil || len(communityEvent.ChannelData.ChannelId) == 0 {
+			return errors.New("invalid community channel delete event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_CHANNEL_EDIT:
+		if communityEvent.ChannelData == nil || len(communityEvent.ChannelData.ChannelId) == 0 ||
+			communityEvent.ChannelData.Channel == nil {
+			return errors.New("invalid community channel edit event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_CHANNEL_REORDER:
+		if communityEvent.ChannelData == nil || len(communityEvent.ChannelData.ChannelId) == 0 {
+			return errors.New("invalid community channel reorder event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_CATEGORY_REORDER:
+		if communityEvent.CategoryData == nil || len(communityEvent.CategoryData.CategoryId) == 0 {
+			return errors.New("invalid community category reorder event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_REQUEST_TO_JOIN_ACCEPT:
+		if len(communityEvent.MembersAdded) == 0 {
+			return errors.New("invalid community request to join accepted event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_REQUEST_TO_JOIN_REJECT:
+		if communityEvent.RejectedRequestsToJoin == nil {
+			return errors.New("invalid community request to join reject event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_MEMBER_KICK:
+		if len(communityEvent.MemberToAction) == 0 {
+			return errors.New("invalid community member kick event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_MEMBER_BAN:
+		if len(communityEvent.MemberToAction) == 0 {
+			return errors.New("invalid community member ban event")
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_MEMBER_UNBAN:
+		if len(communityEvent.MemberToAction) == 0 {
+			return errors.New("invalid community member unban event")
+		}
+
+	default:
```

Comment:
a new event will result in an error from this function, which means no further processing will happen, that makes upgrading the protocol (i.e adding new events), tricky

Reply:
Great point, thank you 👍